apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo3/network/http/HttpUrl.kt

```diff
@@ -0,0 +1,5 @@
+package com.apollographql.apollo3.network.http
+
+expect class HttpUrlBuilder(baseUrl: String, queryParameters: Map<String, String>) {
+  fun build(): String
+}
```

Comment:
is there plans to extend this class in future? It looks a little odd to have class with one fun that returns just a string? Should it be just top level function `buildUrl`? Or  even better introduce `Url / Uri` class that represents url and will have companion factory object?

Reply:
I'd love the Kotlin stlib to ship something there so we can remove this. I'm a bit hesitant on introducing a MPP `Url/Uri` as `Url` pulls a lot of potentially complex things (scheme, hosts, absolute vs relative, etc..) hence the very targetted API. Agreed there's no need for a class though, I'll replace with `buildUrl`